Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM openjdk:8 as builder
+
+RUN apt-get update
+
+# Install npm and nodejs
+RUN apt-get -qq -y install \
+    curl \
+    unzip \
+    build-essential \
+    libssl-dev \
+    libpng-dev \
+    zip
+
+RUN mkdir /root/.nvm
+ENV NVM_DIR /root/.nvm
+ENV NODE_VERSION 10.16.0
+ARG build_env="dev"
+
+RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.34.0/install.sh | bash
+RUN chmod +x $HOME/.nvm/nvm.sh
+RUN . $HOME/.nvm/nvm.sh && nvm install $NODE_VERSION && nvm alias default $NODE_VERSION && nvm use default && npm install -g npm 
+
+RUN ln -sf /root/.nvm/versions/node/v$NODE_VERSION/bin/node /usr/bin/nodejs
+RUN ln -sf /root/.nvm/versions/node/v$NODE_VERSION/bin/node /usr/bin/node
+RUN ln -sf /root/.nvm/versions/node/v$NODE_VERSION/bin/npm /usr/bin/npm
+
+RUN npm install -g webpack webpack-cli yarn 
+
+RUN ln -sf /root/.nvm/versions/node/v$NODE_VERSION/bin/webpack /usr/bin/webpack
+RUN ln -sf /root/.nvm/versions/node/v$NODE_VERSION/bin/webpack-cli /usr/bin/webpack-cli
+
+#
+
+RUN curl -s "https://get.sdkman.io" | bash
+RUN chmod a+x "$HOME/.sdkman/bin/sdkman-init.sh"
+RUN /bin/bash -c "source $HOME/.sdkman/bin/sdkman-init.sh"
+RUN export SDKMAN_DIR="$HOME/.sdkman"
+RUN /bin/bash -c "source $HOME/.sdkman/bin/sdkman-init.sh; sdk install grails 3.3.8"
+RUN /bin/bash -c "source $HOME/.sdkman/bin/sdkman-init.sh; sdk install gradle 3.5"
+RUN /bin/bash -c "source $HOME/.sdkman/bin/sdkman-init.sh; sdk install groovy 2.5.7"
```

Comment:
nit: I think all of these run commands could be one-lined and reduce the image size by a huge amount, but that would be an optimization that could go into a follow-up PR.

Reply:
Makes sense. This is on the builder image, should not impact on final image, I think. Will do it in a next PR.